index.js

```diff
@@ -37,7 +37,7 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
 
   compiler.plugin('make', function (compilation, callback) {
     // Compile the template (queued)
-    compilationPromise = childCompiler.compileTemplate(self.options.template, compiler.context, self.options.filename, compilation)
+    compilationPromise = childCompiler.compileTemplate(self.options.template, compiler.context, 'template', compilation)
```

Comment:
Webpack will overwrite an existing file with the name 'template'. This could also conflict if you would have two instances of the webpack-html-plugin.